modules/light-clients/08-wasm/types/store_test.go

```diff
@@ -1,11 +1,316 @@
 package types_test
 
 import (
+	"errors"
+
 	prefixstore "cosmossdk.io/store/prefix"
+	storetypes "cosmossdk.io/store/types"
 
+	wasmtesting "github.com/cosmos/ibc-go/modules/light-clients/08-wasm/testing"
 	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
 )
 
+var invalidPrefix = []byte("invalid/")
+
+// TestGetStore tests the getStore method of the updateProposalWrappedStore.
+func (suite *TypesTestSuite) TestGetStore() {
+	// calls suite.SetupWasmWithMockVM() and creates two clients with their respective stores
+	subjectStore, substituteStore := suite.GetSubjectAndSubstituteStore()
+
+	testCases := []struct {
+		name     string
+		prefix   []byte
+		expStore storetypes.KVStore
+		expPanic error
+	}{
+		{
+			"success: subject store",
+			types.SubjectPrefix,
```

Comment:
could maybe add a test case with a prefix that has both subject and substitute strings